packages/mobx/__tests__/v5/base/typescript-tests.ts

```diff
@@ -2054,6 +2054,17 @@ test("type inference of the action callback", () => {
     }
 })
 
+test("TS - type inference of Set", () => {
+    const set = observable.set<number>()
+    set.add(1)
+    // @ts-expect-error
+    set.add("1")
+    // @ts-expect-error
+    set.has("1")
+    // @ts-expect-error
+    set.delete("1")
+})
```

Comment:
A nice feature of `ts-expect-error` is that it throws if there is no TS error on the next line